backend/src/database/repositories/tenantRepository.ts

```diff
@@ -239,6 +245,15 @@ class TenantRepository {
   }
 
   static async findById(id, options: IRepositoryOptions) {
+    const redis = await createRedisClient(true)
```

Comment:
Don't create `RedisCache` and `redis` every time - make it static and created lazily upon first call.